### PR TITLE
Updated dependencies and resolving CVEs, bumped SpringBoot to 3.3.5

### PR DESCRIPTION
**Pull request by madduci**
Hello

this PR addresses some CVEs that are present in the code base, due to some libraries.

Updated Libraries:

- SpringBoot 
- Spring Components
- Jackson
- Log4j
- Logback
- Swagger UI
- Bootstrap 
- OpenTelemetry Instrumentation
- Hibernate
- Testcontainers
- MySQL Connector
- PostgreSQL driver
- Oracle11 driver
- ElasticSearch
- GraphQL

### CODE REVIEW

**Comment by jamesagnew** [comment]
Hey @madduci , thanks for the contribution!

Two quick initial comments:

* It looks like this is [not compiling cleanly](https://dev.azure.com/hapifhir/HAPI%20FHIR/_build/results?buildId=16759&view=logs&jobId=d7cd72ff-9a94-5904-ba35-7ed276046804) because of the hibernate search bump. That needs to be investigated.
* Can you please create a changes.yaml changelog outlining the dependency bumps we've made? It should go in the `7_8_0` directory. You could use this one as a template: https://github.com/hapifhir/hapi-fhir/blob/9365ad9509a0cf1c7d6dc228dba7470dd78c188a/hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/changes.yaml

**Comment by madduci** [comment]
Hello and thanks for reviewing it.

I need to perform some corrections, unfortunately it is really slow on my machine. I will apply the requested changes as well.
Michele Adduci

**Comment by jamesagnew** [comment]
@madduci it looks like there is still one still failing CI test: `GraphQLR4Test`. It's failing with a super weird error, I'm having a look to see if I can understand what's up with that.

**Comment by jamesagnew** [comment]
@madduci FYI I fixed that graphql issue and pushed it in.

It turns out that we have a team internally over here who are working on bumping hibernate already (but not boot or - and they've already rules out the `timestampdiff` approach for resolving that one issue because it's not cross-database compatible.

I'm going to propose that we let that work get merged and sync it into your branch just to avoid any extra work - We'll get this merged, but it'll be in a few days probably.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/data/IBatch2WorkChunkRepository.java [hunk 1]
```diff
@@ -150,6 +150,6 @@ List<String> fetchAllChunkIdsForStepWithStatus(
 			@Param("status") WorkChunkStatusEnum theStatus);
 
 	@Query(
-			"SELECT new ca.uhn.fhir.batch2.model.BatchWorkChunkStatusDTO(e.myTargetStepId, e.myStatus, min(e.myStartTime), max(e.myEndTime), avg(e.myEndTime - e.myStartTime), count(*)) FROM Batch2WorkChunkEntity e WHERE e.myInstanceId=:instanceId GROUP BY e.myTargetStepId, e.myStatus")
+			"SELECT new ca.uhn.fhir.batch2.model.BatchWorkChunkStatusDTO(e.myTargetStepId, e.myStatus, min(e.myStartTime), max(e.myEndTime), avg(timestampdiff(SECOND, e.myStartTime, e.myEndTime) * 1000), count(*)) FROM Batch2WorkChunkEntity e WHERE e.myInstanceId=:instanceId GROUP BY e.myTargetStepId, e.myStatus")
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/data/IBatch2WorkChunkRepository.java:153]: issue: I do not believe this is DB-portable across psql/mysql/mssql/oracle/h2. I know mysql and h2 support it, but i do not believe oracle or psql do. Can you change this to the following, which is JPQL-compliant?
```
	@Query(
			"SELECT new ca.uhn.fhir.batch2.model.BatchWorkChunkStatusDTO(e.myTargetStepId, e.myStatus, min(e.myStartTime), max(e.myEndTime), avg(cast((e.myEndTime - e.myStartTime) as long)), count(*)) FROM Batch2WorkChunkEntity e WHERE e.myInstanceId=:instanceId GROUP BY e.myTargetStepId, e.myStatus")
```

Comment (madduci) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/data/IBatch2WorkChunkRepository.java]: got it, thanks for the suggestion

File: pom.xml [hunk 2]
```diff
@@ -1013,53 +1018,54 @@
...
-		<logback_version>1.4.14</logback_version>
+		<flyway_version>10.20.1</flyway_version>
+		<hibernate_version>6.6.2.Final</hibernate_version>
+		<logback_version>1.5.12</logback_version>
 		<!-- Update lucene version when you update hibernate-search version - These go together! -->
-		<hibernate_search_version>7.0.1.Final</hibernate_search_version>
+		<hibernate_search_version>7.2.1.Final</hibernate_search_version>
 		<!-- Update lucene version when you update hibernate-search version - These go together! -->
-		<lucene_version>9.8.0</lucene_version>
+		<lucene_version>9.12.0</lucene_version>
```
Comment (tadgh) [inline pom.xml:1031]: HS 7.2.1 specifically calls out lucene 9.11.1 as a dependency, recommend downgrading this:
- https://hibernate.org/search/releases/7.2/
- https://mvnrepository.com/artifact/org.hibernate.search/hibernate-search-backend-lucene/7.2.1.Final

We have hit problems before when upgrading lucene past the HS-compatible version, see: https://github.com/hapifhir/hapi-fhir/pull/6468/files

**Review by tadgh — APPROVED** [review]
Approved once lucene is downgraded a touch, then we can get this merged in

File: pom.xml (hunk 2 above)
Comment (madduci) [reply pom.xml]: thank you for the advice, i've just pushed the change

**Review by tadgh — APPROVED** [review]

**Comment by tadgh** [comment]
Current test failures occurring: 
```
Could not retrieve code analysis results - Maven run failed.
12:37:31,376 [INFO] Results:
12:37:31,376 [INFO] 
12:37:31,377 [ERROR] Failures: 
12:37:31,378 [ERROR]   FhirResourceDaoR4QueryCountTest.testValidate:668 expected: <10> but was: <9>
12:37:31,378 [ERROR]   FhirResourceDaoR4QueryCountTest.testValueSetExpand_NotPreExpanded_DontUseHibernateSearch:3568 expected: <6> but was: <5>
12:37:31,379 [ERROR]   FhirResourceDaoR4QueryCountTest.testValueSetExpand_NotPreExpanded_UseHibernateSearch:3525 [
 *select tvs1_0.PID,tvs1_0.EXPANSION_STATUS,tvs1_0.EXPANDED_AT,tvs1_0.VSNAME,tvs1_0.RES_ID,tvs1_0.TOTAL_CONCEPT_DESIGNATIONS,tvs1_0.TOTAL_CONCEPTS,tvs1_0.URL,tvs1_0.VER from TRM_VALUESET tvs1_0 join HFJ_RESOURCE rt1_0 on rt1_0.RES_ID=tvs1_0.RES_ID where tvs1_0.URL='http://example.com/my_value_set' order by rt1_0.RES_UPDATED desc fetch first '1' rows only
 * select rt1_0.RES_ID,rt1_0.RES_DELETED_AT,rt1_0.FHIR_ID,rt1_0.RES_VERSION,rt1_0.SP_HAS_LINKS,rt1_0.HAS_TAGS,rt1_0.HASH_SHA256,rt1_0.SP_INDEX_STATUS,rt1_0.RES_LANGUAGE,rt1_0.SP_CMPSTR_UNIQ_PRESENT,rt1_0.SP_CMPTOKS_PRESENT,rt1_0.SP_COORDS_PRESENT,rt1_0.SP_DATE_PRESENT,rt1_0.SP_NUMBER_PRESENT,rt1_0.SP_QUANTITY_NRML_PRESENT,rt1_0.SP_QUANTITY_PRESENT,rt1_0.SP_STRING_PRESENT,rt1_0.SP_TOKEN_PRESENT,rt1_0.SP_URI_PRESENT,rt1_0.PARTITION_DATE,rt1_0.PARTITION_ID,rt1_0.RES_PUBLISHED,rt1_0.RES_TYPE,rt1_0.SEARCH_URL_PRESENT,rt1_0.RES_UPDATED,rt1_0.RES_VER from HFJ_RESOURCE rt1_0 where rt1_0.RES_ID='10833'
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tcsv1_0.PID,tcsv1_0.CODESYSTEM_PID,tcsv1_0.CS_DISPLAY,tcsv1_0.CS_VERSION_ID,tcsv1_0.RES_ID from TRM_CODESYSTEM_VER tcsv1_0 where tcsv1_0.PID='108'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,md1_0.CONCEPT_PID,md1_0.PID,md1_0.CS_VER_PID,md1_0.LANG,md1_0.USE_CODE,md1_0.USE_DISPLAY,md1_0.USE_SYSTEM,md1_0.VAL,md1_0.VAL_VC,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 left join TRM_CONCEPT_DESIG md1_0 on tc1_0.PID=md1_0.CONCEPT_PID where tc1_0.PID in ('734','735','736','737','738','739','740')] 
expected: 6
 but was: 5
12:37:31,379 [ERROR]   ResourceProviderR4ValueSetNoVerCSNoVerTest.testExpandUsingHierarchy_NotPreStored:1114 [select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tcsv1_0.PID,tcsv1_0.CODESYSTEM_PID,tcsv1_0.CS_DISPLAY,tcsv1_0.CS_VERSION_ID,tcsv1_0.RES_ID from TRM_CODESYSTEM_VER tcsv1_0 where tcsv1_0.PID='520'
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 where tc1_0.CODESYSTEM_PID='520' and tc1_0.CODEVAL='A'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,md1_0.CONCEPT_PID,md1_0.PID,md1_0.CS_VER_PID,md1_0.LANG,md1_0.USE_CODE,md1_0.USE_DISPLAY,md1_0.USE_SYSTEM,md1_0.VAL,md1_0.VAL_VC,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 left join TRM_CONCEPT_DESIG md1_0 on tc1_0.PID=md1_0.CONCEPT_PID where tc1_0.PID in ('12486','12488','12487')
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tcsv1_0.PID,tcsv1_0.CODESYSTEM_PID,tcsv1_0.CS_DISPLAY,tcsv1_0.CS_VERSION_ID,tcsv1_0.RES_ID from TRM_CODESYSTEM_VER tcsv1_0 where tcsv1_0.PID='520'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,md1_0.CONCEPT_PID,md1_0.PID,md1_0.CS_VER_PID,md1_0.LANG,md1_0.USE_CODE,md1_0.USE_DISPLAY,md1_0.USE_SYSTEM,md1_0.VAL,md1_0.VAL_VC,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 left join TRM_CONCEPT_DESIG md1_0 on tc1_0.PID=md1_0.CONCEPT_PID where tc1_0.PID in ('12485')] 
expected: 10
 but was: 8
12:37:31,379 [ERROR]   ResourceProviderR4ValueSetNoVerCSNoVerTest.testExpandUsingHierarchy_PreStored_NotPreCalculated:1074 [SELECT t1.RES_ID FROM HFJ_RESOURCE t1 INNER JOIN HFJ_SPIDX_URI t0 ON (t1.RES_ID = t0.RES_ID) WHERE (t0.HASH_URI = '-3219147991190349831') ORDER BY t1.RES_UPDATED DESC NULLS LAST fetch first '1' rows only
 * select rsv1_0.PID,rsv1_0.RES_DELETED_AT,rsv1_0.RES_ENCODING,rsv1_0.FHIR_ID,rsv1_0.RES_VERSION,rsv1_0.HAS_TAGS,rsv1_0.PARTITION_ID,rsv1_0.PROV_REQUEST_ID,rsv1_0.PROV_SOURCE_URI,rsv1_0.RES_PUBLISHED,rsv1_0.RES_TEXT,rsv1_0.RES_ID,rsv1_0.RES_TEXT_VC,rsv1_0.RES_TYPE,rsv1_0.RES_VER,rsv1_0.RES_UPDATED from ( SELECT h.PID               as PID,                           r.RES_ID            as RES_ID,                        h.RES_TYPE          as RES_TYPE,                      h.RES_VERSION       as RES_VERSION,                   h.RES_VER           as RES_VER,                       h.HAS_TAGS          as HAS_TAGS,                      h.RES_DELETED_AT    as RES_DELETED_AT,                h.RES_PUBLISHED     as RES_PUBLISHED,                 h.RES_UPDATED       as RES_UPDATED,                   h.RES_TEXT          as RES_TEXT,                      h.RES_TEXT_VC       as RES_TEXT_VC,                   h.RES_ENCODING      as RES_ENCODING,                  h.PARTITION_ID      as PARTITION_ID,                  p.SOURCE_URI        as PROV_SOURCE_URI,               p.REQUEST_ID        as PROV_REQUEST_ID,               r.FHIR_ID         as FHIR_ID      FROM HFJ_RESOURCE r     INNER JOIN HFJ_RES_VER h ON r.RES_ID = h.RES_ID and r.RES_VER = h.RES_VER    LEFT OUTER JOIN HFJ_RES_VER_PROV p ON p.RES_VER_PID = h.PID  ) rsv1_0 where rsv1_0.RES_ID in ('27784')
 * select tvs1_0.PID,tvs1_0.EXPANSION_STATUS,tvs1_0.EXPANDED_AT,tvs1_0.VSNAME,tvs1_0.RES_ID,tvs1_0.TOTAL_CONCEPT_DESIGNATIONS,tvs1_0.TOTAL_CONCEPTS,tvs1_0.URL,tvs1_0.VER from TRM_VALUESET tvs1_0 join HFJ_RESOURCE rt1_0 on rt1_0.RES_ID=tvs1_0.RES_ID where tvs1_0.URL='http://example.com/my_value_set' order by rt1_0.RES_UPDATED desc fetch first '1' rows only
 * select rt1_0.RES_ID,rt1_0.RES_DELETED_AT,rt1_0.FHIR_ID,rt1_0.RES_VERSION,rt1_0.SP_HAS_LINKS,rt1_0.HAS_TAGS,rt1_0.HASH_SHA256,rt1_0.SP_INDEX_STATUS,rt1_0.RES_LANGUAGE,rt1_0.SP_CMPSTR_UNIQ_PRESENT,rt1_0.SP_CMPTOKS_PRESENT,rt1_0.SP_COORDS_PRESENT,rt1_0.SP_DATE_PRESENT,rt1_0.SP_NUMBER_PRESENT,rt1_0.SP_QUANTITY_NRML_PRESENT,rt1_0.SP_QUANTITY_PRESENT,rt1_0.SP_STRING_PRESENT,rt1_0.SP_TOKEN_PRESENT,rt1_0.SP_URI_PRESENT,rt1_0.PARTITION_DATE,rt1_0.PARTITION_ID,rt1_0.RES_PUBLISHED,rt1_0.RES_TYPE,rt1_0.SEARCH_URL_PRESENT,rt1_0.RES_UPDATED,rt1_0.RES_VER from HFJ_RESOURCE rt1_0 where rt1_0.RES_ID='27784'
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tcsv1_0.PID,tcsv1_0.CODESYSTEM_PID,tcsv1_0.CS_DISPLAY,tcsv1_0.CS_VERSION_ID,tcsv1_0.RES_ID from TRM_CODESYSTEM_VER tcsv1_0 where tcsv1_0.PID='515'
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 where tc1_0.CODESYSTEM_PID='515' and tc1_0.CODEVAL='A'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,md1_0.CONCEPT_PID,md1_0.PID,md1_0.CS_VER_PID,md1_0.LANG,md1_0.USE_CODE,md1_0.USE_DISPLAY,md1_0.USE_SYSTEM,md1_0.VAL,md1_0.VAL_VC,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 left join TRM_CONCEPT_DESIG md1_0 on tc1_0.PID=md1_0.CONCEPT_PID where tc1_0.PID in ('12383','12385','12384')
 * select tcs1_0.PID,tcs1_0.CODE_SYSTEM_URI,tcs1_0.CURRENT_VERSION_PID,tcs1_0.CS_NAME,tcs1_0.RES_ID from TRM_CODESYSTEM tcs1_0 where tcs1_0.CODE_SYSTEM_URI='http://example.com/my_code_system'
 * select tcsv1_0.PID,tcsv1_0.CODESYSTEM_PID,tcsv1_0.CS_DISPLAY,tcsv1_0.CS_VERSION_ID,tcsv1_0.RES_ID from TRM_CODESYSTEM_VER tcsv1_0 where tcsv1_0.PID='515'
 * select tc1_0.PID,tc1_0.CODEVAL,tc1_0.CODESYSTEM_PID,md1_0.CONCEPT_PID,md1_0.PID,md1_0.CS_VER_PID,md1_0.LANG,md1_0.USE_CODE,md1_0.USE_DISPLAY,md1_0.USE_SYSTEM,md1_0.VAL,md1_0.VAL_VC,tc1_0.DISPLAY,tc1_0.INDEX_STATUS,tc1_0.PARENT_PIDS,tc1_0.PARENT_PIDS_VC,tc1_0.CODE_SEQUENCE,tc1_0.CONCEPT_UPDATED from TRM_CONCEPT tc1_0 left join TRM_CONCEPT_DESIG md1_0 on tc1_0.PID=md1_0.CONCEPT_PID where tc1_0.PID in ('12382')] 
expected: 14
 but was: 12

```

@madduci  can you run these locally and see if they pass for you? It's possible but unlikely that these are intermittents.

**Comment by madduci** [comment]
@tadgh actually i have the same error, but for DSTU3, instead of R4:

image

**Comment by jamesagnew** [comment]
Hi @madduci - I've committed a couple of additional fixes needed because of the Hibernate version bump (we have some downstream tests that caught the issues). Assuming everything passes, this should merge tonight or tomorrow.

**Comment by madduci** [comment]
Thank you very much for your help @jamesagnew !

**Comment by Romanow88** [comment]
Hey @jamesagnew, can these updates (especially the Hibernate update and the CVEs) be merged to a release branch (7.6.2 or something of the like)?

**Pull request merged**